knowledge_graphs/parse_repo_into_neo4j.py

```
@@ -0,0 +1,280 @@
+                
+                file_info = {
+                    "path": rel_path,
+                    "name": file,
+                    "extension": file_ext,
+                    "size": os.path.getsize(file_path)
+                }
+                
+                # Extract content for supported file types
+                if file_ext in code_extensions or file_ext in doc_extensions or file_ext in config_extensions:
+                    try:
+                        with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
```

Ruff (0.12.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)